code-challenges.test.js

```diff
@@ -16,35 +16,120 @@
+// const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+// // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+// //      Test Suites: 1 passed, 1 total
+// //      Tests:       2 passed, 2 total
+
+// // b) Create the function that makes the test pass.
+// // Input: [array] of words
+// // Output: first word removed and words returned reshuffled
+// // PsuedoCode:
+// //  Create a function that takes in an array of words.
+// const removeandreShuffle = (arr) => {
+// // Create a for loop to iterate through the array of words
+//     for (let i = 0; i < arr.length; i++) {
+// // Create a variable to save word in current index.
+//         let holdWord = arr[1]
+// // Create a variable and assign a random generator to to identify word
+// // in array.
+//         let randomIndex = Math.floor(Math.random(arr.length))
+// // Word saved in array index is replaced with word from 
+// // random word generator.
+//         arr[i] = arr[randomIndex]
+// // Replacement word will be returned new array.
+//         arr[randomIndex] = holdWord
+//     }
+// // Return first word removed and words returned reshuffled.
+//     return arr
+// }
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
 
+// --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
-// b) Create the function that makes the test pass.
+// a) Create a test with expect statements for each of the variables provided.
 
+///     it("contains up votes and down votes and returns the end tally", () => {
+//       expect(voteTally(votes1)).toEqual("11")
+//     })
+//   })
 
-// --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
+//   // Create a function and have test, run the test and practice
+//   // the red step in the red-green-refactor.  
+// const voteTally = (obj) => {
+//     return Object.values(obj).parseInt()
+//     console.log(values)
+//    }
 
-// a) Create a test with expect statements for each of the variables provided.
+// // Test Suites: 1 failed, 1 total
+// // Tests:       1 failed, 1 total
+
+describe("endtallyVotes", () => {
+  it("contains up votes and down votes and returns the end tally", () => {
+    expect(endtallyVotes(votes1)).toEqual(11)
+  })
+})
 
 const votes1 = {upVotes: 13, downVotes: 2}
-// Expected output: 11
+// // Expected output: 11
 const votes2 = {upVotes: 2, downVotes: 33}
 // Expected output: -31
 
+// 23 Nov.-- Mistake found...
+const endtallyVotes = (object) => {
+  // Create a function, that takes an object
+  return object.upVotes - object.downVotes
+  // oject can define own behavior, called instance methods. Called through type
+  // conversion and you don't have to call yourself in your code.
```

Comment:
Your code and logic looks great here.  Accessing a value in an object is not type coercion though.  Type coercion would be turning a number into a string due to implied evaluations like 5 == '5'.